module/VuFind/tests/unit-tests/src/VuFindTest/Search/Solr/ResultsTest.php

```diff
@@ -420,6 +420,259 @@ public function testGetFacetList(): void
         $facets = $resultsNoHelper->getFacetList(['building' => 'Building']);
     }
 
+    /**
+     * Test exclude filters
+     *
+     * @return void
+     */
+    public function testExcludeFilters()
+    {
+        $config = $this->getMockConfigPluginManager(
+            [
+                'facets' => [
+                    'SpecialFacets' => [
+                        'hierarchical' => [
+                            'building',
+                            'format',
+                        ],
+                    ],
+                    'HierarchicalExcludeFilters' => [
+                        'format' => [
+                            '1/Book/AudioBook/',
+                        ],
+                    ],
+                ],
+            ]
+        );
+
+        $results = $this->getResultsFromResponse(
+            [
+                'response' => [
+                    'numFound' => 7,
+                ],
+                'facet_counts' => [
+                    'facet_fields' => [
+                        'building' => [
+                            ['0/Helsinki/', 25],
+                            ['1/Helsinki/NatLibFi/', 15],
+                            ['1/Helsinki/Library/', 15],
+                        ],
+                        'format' => [
+                            ['0/Book/', 123],
+                            ['1/Book/AudioBook/', 25],
+                            ['1/Book/EBook/', 12],
+                            ['1/Book/ColoringBook/', 12],
+                        ],
+                    ],
+                ],
+            ],
+            $this->getParams(null, $config)
+        );
+
+        $results->setHierarchicalFacetHelper(new HierarchicalFacetHelper());
```

Comment:
I didn't study this in great depth, so maybe there's a strong reason to keep these tests in this place -- but this seems more like an integration test than a unit test, since it's testing the HierarchicalFacetHelper through the Solr Results object. Would it be better to do this testing in the [HierarchicalFacetHelper test](https://github.com/vufind-org/vufind/blob/dev/module/VuFind/tests/unit-tests/src/VuFindTest/Search/Solr/HierarchicalFacetHelperTest.php) and test here with a simple mock to confirm that the Results object is passing along the values as expected?

If this is a major ordeal to refactor, please don't feel obligated to change, but if it's not too hard, it might keep our test suite a little better organized.

Reply:
sounds good! I was also a littlebit lost as to where I should put this test but im going to try and adjust.